### PR TITLE
bugfix/#3267

### PR DESCRIPTION
**Pull request by KKatiaa**
Event search [#3267](https://github.com/ita-social-projects/GreenCity/issues/3267)

### CODE REVIEW

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 1]
```diff
@@ -107,6 +115,29 @@ export class EventsListComponent implements OnInit, OnDestroy {
...
+  searchWords(): void {
+    this.searchFilterWords.valueChanges.subscribe((value) => {
+      this.wordsToSearch = value.split(' ');
+      this.sortByWord(this.bufferArray, this.wordsToSearch);
+    });
+  }
+
+  sortByWord(list: EventPageResponceDto[], words: string[]): void {
+    this.eventsList = list.filter((element) => {
+      for (const word of words) {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts:139]: use forEach here

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 2]
```diff
@@ -107,6 +115,29 @@ export class EventsListComponent implements OnInit, OnDestroy {
...
 
+  searchWords(): void {
+    this.searchFilterWords.valueChanges.subscribe((value) => {
+      this.wordsToSearch = value.split(' ');
+      this.sortByWord(this.bufferArray, this.wordsToSearch);
+    });
+  }
+
+  sortByWord(list: EventPageResponceDto[], words: string[]): void {
+    this.eventsList = list.filter((element) => {
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts:138]: this.eventsList = list.filter(item => words.includes(item.title))

File: src/app/main/component/events/components/events-list/events-list.component.ts [hunk 3]
```diff
@@ -107,6 +115,29 @@ export class EventsListComponent implements OnInit, OnDestroy {
...
+      for (const word of words) {
+        if (element.title.includes(word)) {
+          return true;
+        }
+      }
+      return false;
+    });
+    this.eventsList.length ? (this.noEventsMatch = false) : (this.noEventsMatch = true);
```
Comment (hnativlyubomyr) [inline src/app/main/component/events/components/events-list/events-list.component.ts:146]: this.noEventsMatch =  !!this.eventsList.length

Comment (KKatiaa) [reply src/app/main/component/events/components/events-list/events-list.component.ts:146]: changed it

**Review by OksanaFilipchuk — APPROVED** [review]

**Review by hnativlyubomyr — APPROVED** [review]

**Pull request merged**